server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregatorFactory.java

```diff
@@ -47,4 +48,10 @@ public Aggregator createInternal(Aggregator parent,
         return FiltersAggregator.build(name, factories, filters, keyed,
             otherBucket ? otherBucketKey : null, context, parent, cardinality, metadata);
     }
+
+    @Override
+    public Set<String> fieldsUsed() {
+        return Set.of();
```

Comment:
The PR should cover all fields which are "aggregated on". What's missing is accesses that are typically associated with a query context (e.g. filter(s) aggregation), and which I would like to classify differently (those will be equated to access as part of a query).

Reply:
That makes sense.  Let's just leave a comment on these cases so in six months I remember why we did it this way, please?

Reply:
I've extended the javadocs on the `fieldsUsed` method in dbb5252. In a follow-up PR, all this will become clearer as well as I'll be adding another method next to `fieldsUsed()` called `queriesUsed()` that returns the queries that were used (applicable e.g. on filter(s) aggregations)